megatron/training.py

```diff
@@ -57,6 +57,49 @@
 from megatron.model.gpt2_model import cross_entropy
 from eval_tasks import run_eval_harness
 
+def save_base_shapes(neox_args, base_shapes, use_cache):
+    from copy import deepcopy
+    neox_args_copy = deepcopy(neox_args)
+
+    # Instantiation of the base model fails in the init function (init_functions.py) because we haven't called set_base_shapes on it at this point, so disable it temporarily here
+    neox_args.use_mup = False
+
+    base_model = GPT2ModelPipe(
+                    neox_args=neox_args,
+                    num_tokentypes=0,
+                    parallel_output=True,
+                    topology=mpu.get_topology(),
+                    use_cache=use_cache)
+
+    if not neox_args.is_pipe_parallel:
+        base_model = base_model.to_sequential()
+
+    base_shapes = mup.get_shapes(base_model)
+
+    del base_model
+
+    neox_args.hidden_size = neox_args.hidden_size * 2
```

Comment:
I think if megatron/neox_arguments/neox_args.py supports taking in a list and we identified which hparams are good candidates for scaling, it wouldn't be too much effort to edit the code for that.

Also, what you set each hparam to in the delta model might not matter. From the docs it just says you need to change whatever you want to scale. It might be worth double checking that with the authors though.

Reply:
Ok. Let's revisit this once the PR is more mature.

Reply:
@nsarka @Quentin-Anthony now that the PR is mature, we should revisit this conversation.

Reply:
@StellaAthena -- We discussed this offline and added a new arg `mup_width_scale` to scale the hidden size (default = 2). If we find scaling other hparams provides benefits we can add those in the future.